### ISSUE TITLE
Infinite loop with Jsoup 1.9.1

### ISSUE THREAD

**Issue by jbflanigan** [jhy/jsoup#704]
We recently upgraded from 1.8.3 to 1.9.1 and noticed many instances of threads looping inside TokeniserState.handleDataEndTag(). I believe this occurs because CharacterReader.matchesLetter() checks non-ascii letters via Character.isLetter(c), but consumeLetterSequence() does not. 

Here's a snippet that reproduces the problem. The HTML is obviously not quite valid, but it represents the sort of content we encounter in the wild.

```java
String html = "<html><body><style>bogus</styleá><p>just a small test page</p></body></html>";
Jsoup.parseBodyFragment(html);
```

**Comment by jhy** [comment]
Fixed with 49db84f8f2beb846fb8271fa20a28271afe37d90, thanks!

**Comment by jhy** [comment]
Whoops - corrected test - 2bca40cf59bfda485f2fbf43b8bd66509a8f98ab
